cmd/syncthing/gui.go

```diff
@@ -652,10 +652,14 @@ func restPostUpgrade(w http.ResponseWriter, r *http.Request) {
 func restPostScan(m *model.Model, w http.ResponseWriter, r *http.Request) {
 	qs := r.URL.Query()
 	folder := qs.Get("folder")
-	sub := qs.Get("sub")
-	err := m.ScanFolderSub(folder, sub)
-	if err != nil {
-		http.Error(w, err.Error(), 500)
+	if folder != "undefined" {
+		sub := qs.Get("sub")
+		err := m.ScanFolderSub(folder, sub)
+		if err != nil {
+			http.Error(w, err.Error(), 500)
+		}
+	} else {
+		m.ScanFolders()
```

Comment:
We should perhaps change ScanFolder to return errors (if any), and do a 500 if we've got an error.
Though I don't care too much about this.